salt/states/junos.py

```diff
@@ -33,16 +33,15 @@ def wrapper(*args, **kwargs):
 
 
 @resultdecorator
-def rpc(name, dest=None, format="xml", args=None, **kwargs):
+def rpc(name, **kwargs):
```

Comment:
Salt initially made changes in Junos-salt modules [https://github.com/saltstack/salt/commit/93ee5ee2b01b6f752f5538d0da413719d410d65c#diff-15ad308a12ea5b6d5188dd4a94eab4a0R490 https://github.com/saltstack/salt/commit/93ee5ee2b01b6f752f5538d0da413719d410d65c#diff-15ad308a12ea5b6d5188dd4a94eab4a0R400]. This commit caused all our state module to break [https://github.com/Juniper/salt/issues/77]. 

@dmurphy18  Hi David, The issue was caused as salt changed positional arguments(*args) in our modules function definition to **kwargs. But, the state modules were not modified in those commit. we had changed it to **kwargs to keep it in line with changes done by salt. 
Also we have neither changes any function name of removed any parameters, just that they are consumed via **kwargs.

Reply:
@vnitinv I checked back as far as 2017.7.0 and the function signature is the same as that remove by your change. I can understand that this may not line up with Juniper's own functionality but it is an interface which has been published for a few years and it MUST go through a deprecation cycle before it can be changed.  And I understand that the kwargs can handle the changes but the positional arguments MUST remain until the deprecation cycle for an interface change has completed. Note can add a new interface and mark the old interface for deprecation.  Existing customers cannot have the rug pulled out from underneath them without time to adjust.

Reply:
@dmurphy18 I have reverted to old function definition itself.